### ISSUE TITLE
Multiple modules in a test file get treated as one

### ISSUE THREAD

**Issue by s3cur3** [carsdotcom/unverified_mox#2]
In our codebase at @felt, we have a pattern of breaking tests that require `async: false` into their own module within the same test file, so that the bulk of the unit tests for a given module can still be `async: true`. What we noticed is that if one of those modules correctly verifies Mox expectations but the other does not, CredoMox will incorrectly give the whole file a pass.

Here's a failing test that demonstrates the issue:

```elixir
    test "warns about files with multiple modules where one module is missing verify_on_exit" do
      """
      defmodule CredoSampleModule1Test do
        import Mox
        setup :verify_on_exit!
        describe "something" do
          test "the thing" do
            expect MockModule, :function, fn -> :foo end
          end
        end
      end

      defmodule CredoSampleModule2Test do
        import Mox
        describe "something" do
          test "the thing" do
            expect MockModule, :function, fn -> :foo end
            expect MockModule, :function, fn -> :foo end
            expect MockModule, :function, fn -> :foo end
          end
        end
      end
      """
      |> to_source_file()
      |> run_check(UnverifiedMox)
      |> assert_issue(fn issue -> assert issue.trigger == "Missing verify_on_exit!" end)
    end
```

I'll try to look into a fix for this in the upcoming week and open a PR.

**Comment by zkayser** [comment]
@s3cur3 Thank you again for raising this Issue and also contributing some nice improvements to the docs and dependency declarations! 

I added the failing test you provided and have a direction for cleaning this up so we do not inadvertently mark files as passing when they should not. I will try to get a PR up and publish a release that addresses the issue in the next day or two.

**Comment by zkayser** [comment]
Okay, I just published a new release (v0.1.1) with the changes from https://github.com/carsdotcom/credo_mox/pull/3. We published a new version on hex [here](https://hexdocs.pm/credo_mox/) this morning. 

Thank you again for the additions and enhancements, and also for the regression test! That made it much easier to work through the change and make sure we got the fix right.

**Comment by s3cur3** [comment]
OMG, that's amazing! Thanks so much! And, er... sorry for not putting up a fix myself. My good intentions didn't carry me as far as I'd hoped... 😖
